### PR TITLE
Allow array values for attributes

### LINKED ISSUE #142: Allow array values for attributes
Add array variant to `Value` enum, see [the spec change](https://github.com/open-telemetry/opentelemetry-specification/pull/368) for details.

### PR DESCRIPTION
**Pull request by siyopao**
The spec says that the string value of the array should look like a JSON string, so I attempted that here. And, as I mentioned earlier, this will allow for non-homogeneous arrays so some there is something to do there. Please let me know if any other changes required. Thank you!

Closes #142.

### CODE REVIEW

**Comment by siyopao** [comment]
Ah, I may need some help with Jaeger. I am not quite sure how to edit those files.

**Comment by jtescher** [comment]
You are correct Jaeger does not support arrays. Looks like the go implementation simply ignores arrays when exporting to jaeger https://github.com/open-telemetry/opentelemetry-go/blob/master/exporters/trace/jaeger/jaeger.go#L281-L328

**Comment by jtescher** [comment]
@siyopao maybe just serialize as Jaeger string tag for now?

**Comment by siyopao** [comment]
Hi @jtescher. Sure, let me try that. I need to figure out how Thrift works, but will try soon. Thanks.

**Comment by jtescher** [comment]
@siyopao Sounds good. Can look at https://github.com/open-telemetry/opentelemetry-rust/blob/master/opentelemetry-jaeger/src/lib.rs#L296-L304 to see how string values are converted currently.

**Comment by siyopao** [comment]
Thank you for the pointer @jtescher.

File: src/api/correlation/propagation.rs [hunk 1]
```diff
@@ -215,6 +215,19 @@ mod tests {
...
+        // "values of array types"
+            (
+                vec![
+                    KeyValue::new("key1", Value::Array(vec![Value::Bool(true), Value::Bool(false)])),
+                    KeyValue::new("key2", Value::Array(vec![Value::String("val1".to_string()), Value::String("val2".to_string())])),
+                    KeyValue::new("key3", Value::Array(vec![Value::I64(123), Value::I64(456)])),
+                ],
+                vec![
+                        "key1=[true%2Cfalse]",
+                        "key2=[val1%2Cval2]",
```
Comment (jtescher) [inline src/api/correlation/propagation.rs:227]: Think the spec wants these to be json encoded values, so strings should probably be quoted here even though it complicates things a little.

**Comment by jtescher** [comment]
@siyopao looks great, just one last tweak and we should be good!

File: src/api/correlation/propagation.rs (hunk 1 above)
Comment (siyopao) [reply src/api/correlation/propagation.rs]: Oops, sorry about that. Fixed.

**Pull request merged**